prometheus_eaton_ups_exporter.py

```diff
@@ -117,37 +118,40 @@ def split_listen_address(listen_address):
     return host_address, port
 
 
-def run(args):
+def run(args: argparse.Namespace) -> None:
     """Execute the Prometheus Eaton UPS Exporter."""
     parser = create_parser()
-    try:
-        args = parser.parse_args(args)
-
-        listen_address = args.__getattribute__('web.listen_address')
-        host_address, port = split_listen_address(listen_address)
-
-        REGISTRY.register(
-            UPSMultiExporter(
-                args.config,
-                insecure=args.insecure,
-                verbose=args.verbose,
-                threading=args.threading,
-                login_timeout=args.login_timeout
-            )
+    args = parser.parse_args(args)
+
+    listen_address = args.__getattribute__('web.listen_address')
+    host_address, port = split_listen_address(listen_address)
+
+    REGISTRY.register(
+        UPSMultiExporter(
+            args.config,
+            insecure=args.insecure,
+            verbose=args.verbose,
+            threading=args.threading,
+            login_timeout=args.login_timeout
         )
+    )
+    # Start up the server to expose the metrics.
+    try:
+        start_http_server(port=int(port), addr=host_address)
+    except OSError as err:
+        if args.verbose:
+            print(traceback.format_exc())
+        else:
+            print(err)
+        sys.exit(1)
 
-        # Start up the server to expose the metrics.
-        start_http_server(int(port), addr=host_address)
-        print(
-            f"Starting Prometheus Eaton UPS Exporter on {host_address}:{port}"
-        )
-        # Run forever until an Error Event or Keyboard Interrupt
+    print(f"Starting Prometheus Eaton UPS Exporter on {host_address}:{port}")
+
+    # Run forever until an Error Event or Keyboard Interrupt
+    try:
```

Comment:
Maybe this is an idiom I don't understand, but I do not see how `while True` and `sleep` can raise anything.

Reply:
It is not raised from that line, but while running that line, when CTRL+C is used, and I tested this, it works like intended.